internal/server/recon/recognize.go

```diff
@@ -73,6 +73,138 @@ func RecognizePlaces(
 	return resp, nil
 }
 
+// RecognizePlaces implements API for Mixer.RecognizePlaces.
+func RecognizeEntities(
+	ctx context.Context,
+	in *pb.RecognizePlacesRequest,
+	store *store.Store,
+	resolveBogusName bool,
+) (*pb.RecognizePlacesResponse, error) {
+
+	resp := &pb.RecognizePlacesResponse{
+		QueryItems: map[string]*pb.RecognizePlacesResponse_Items{},
+	}
+	for _, query := range in.GetQueries() {
+		id2spans := getId2Span(query)
+		if id2spans == nil {
+			continue
+		}
+		idsToResolve := []string{}
+		for id := range id2spans {
+			idsToResolve = append(idsToResolve, id)
+		}
+		resolvedIdEntities, err := GetResolvedIdEntities(ctx, store, "reconName", idsToResolve, "dcid")
+		if err != nil {
+			continue
+		}
+
+		// go through the resolved entities and create a map of spans to the entity
+		// it resolved to
+		span2item := map[string]*pb.RecognizePlacesResponse_Item{}
+		for _, entity := range resolvedIdEntities.GetEntities() {
+			places := []*pb.RecognizePlacesResponse_Place{}
+			for _, id := range entity.GetOutIds() {
+				// TODO: add types to the response
+				places = append(places, &pb.RecognizePlacesResponse_Place{Dcid: id})
+			}
+			for span := range id2spans[entity.GetInId()] {
+				span2item[span] = &pb.RecognizePlacesResponse_Item{Span: span, Places: places}
+			}
+		}
+
+		// get the list of resolved spans and sort them by longer span first
+		spans := []string{}
+		for span := range span2item {
+			spans = append(spans, span)
+		}
+		sort.Slice(spans, func(i, j int) bool {
+			return len(spans[i]) > len(spans[j])
+		})
+
+		// iterate through the resolved spans and add the resolved items to the
+		// response
+		queryRespItems := []*pb.RecognizePlacesResponse_Item{{Span: query}}
+		for _, span := range spans {
```

Comment:
The logic in this loop is hard to follow.  Can you comment "why" we're doing what we're doing with some examples?

Reply:
good idea, done